konan_sdk/endpoints/konan_endpoints.py

```diff
@@ -175,3 +182,60 @@ def process_response(
             endpoint_response.json['failure'],
             endpoint_response.json['total']
         )
+
+
+class CreateDeploymentEndpoint(
+    KonanBaseGenericDeploymentsEndpoint[
+        KonanDeploymentCreationRequest,
+        KonanDeploymentCreationResponse
+    ]
+):
+    @property
+    def name(self) -> str:
+        return 'create-deployment'
+
+    def prepare_request(
+        self, request_object: KonanDeploymentCreationRequest
+    ) -> KonanEndpointRequest:
+        return KonanEndpointRequest(data={
+            'name': request_object.name,
+            'docker_username': request_object.docker_credentials.username,
+            'docker_password': request_object.docker_credentials.password,
+            'image_url': request_object.docker_image.url,
+            'exposed_port': request_object.docker_image.exposed_port,
+        })
+
+    def process_response(
+        self, endpoint_response: KonanEndpointResponse
+    ) -> KonanDeploymentCreationResponse:
+        return KonanDeploymentCreationResponse(
+            KonanDeployment(
+                endpoint_response.json['deployment']['uuid'],
+                endpoint_response.json['deployment']['name'],
+                datetime.datetime.fromisoformat(
+                    endpoint_response.json['deployment']['created_at'],
+                )
+            ),
+            [
+                KonanDeploymentError(
+                    KonanDeploymentErrorType(error['field']),
+                    error['message'],
+                ) for error in endpoint_response.json['errors']
+            ],
+            endpoint_response.json['container_logs']
+        )
+
+    # TODO: Currently, every endpoint class exposes BOTH .get() and .post()
+    # TODO: Perhaps it is better to explicitly support both or split them
```

Comment:
I agree, we need to rethink how we approach this. I am thinking of leaving the Endpoints classes as our sort of atomic unit where they only support 1 http method implementation and create another class for grouping functionality of multiple methods but I am not sure, will require some thought